src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,63 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> head;
+    private Node<T> tail;
```

Comment:
Since the `Node` only has one reference to the other `Node`, we only need either `head` or `tail`, having both is redundant